### DISCUSSION TITLE
Make it easier to create custom utilities that make use of the Tailwind color palette

### DISCUSSION THREAD

**Discussion by Daiz** [tailwindlabs/tailwindcss#6824] [Ideas]
I was trying to create a `text-glow` utility (utilizing text-shadow) and wanted to try making it so that you could do eg. `text-glow-blue-400/25` or use any kind of Tailwind color in general. However, the documentation was of next to no help with this - clearly, a custom plugin utilizing `matchUtilities` was the most appropriate tool for this, but there was no info whatsoever on how to build a custom utility that would support the Tailwind palette. I also took a look at the actual Tailwind code, but that seems to use a lot of internal helper functions to create these kind of utilities, which aren't exposed to users (or at least are not documented in any way).

Anyway, I gave up eventually and settled on a pretty lackluster CSS variable solution instead in which I can't even use `[--color:theme('color.blue.400')]` because the variable is fed to a CSS `rgba()` function and theme only returns hex which doesn't work there.

So with that said, it sure would be nice if there was an easy, streamlined and documented way to create custom utilities that made use of the Tailwind color palette.

**Comment by adamwathan** [comment]
Hey! Yeah haven't had a chance to document some of this more advanced stuff and certain things I think need to be made a little bit easier (like flattening the color palette) but here's an example you can look at:

https://play.tailwindcss.com/IIgMqUampi?file=config

The `type` option is the trick to making the color opacity stuff work.

**Comment by Daiz** [comment]
Thanks! I managed to got it working from there, though a follow-up thought from also implementing glow size control alongside color control (think `text-glow-2 text-glow-red-400`, where larger glows are achieved through multiple shadows ala `text-shadow: 0 0 10px rgba(red-400, 1), 0 0 20px rgba(red-400, 0.5)`.

You can probably see the key issue at hand from above: To be able to make the multiple shadows with opacity control work, I needed a way to tweak the alpha of the incoming color values. I managed to make use of the `parseColor` function from the Tailwind utils to achieve this, but if I had to make a guess, this basically means that first TW is parsing into that internally, then converting to text for the `value` param, then I'm parsing it again from there... It works, but it would certainly be more convenient if I could get a color object instead of a string as the `value` in the first place... though then I guess there is the question of how exactly would all that work with other values in the mix (as I ended up doing `{...flattenColorPalette(theme("colors")), ...theme("textGlow")}` for the matchUtilities `values` for `text-glow`). I imagine it might be somewhat complicated to come up with something elegant for this kind of multiple value scenario, but well, at the very least, I imagine all this could be something to write documentation about in any case.
